client/client.go

```diff
@@ -303,8 +303,9 @@ func (r *NotaryRepository) RemoveTarget(targetName string) error {
 	return nil
 }
 
-// ListTargets lists all targets for the current repository
-func (r *NotaryRepository) ListTargets() ([]*Target, error) {
```

Comment:
If a target gets signed with the same name under different roles (and different hashes), how does the caller do conflict resolution?

The name is something like 'latest', 'edge' or '2.6', so there is nothing that distinguishes which role this particular target came from after calling ListTargets. And since this is a list, the outcome will simply be an array with several 'latest' targets in it.